repos/system_upgrade/common/actors/initramfs/checkinitramfstasks/libraries/checkinitramfstasks.py

```diff
@@ -22,51 +22,65 @@ def _printable_modules(conflicts):
     return ''.join(output)
 
 
-def _treat_path(dmodule):
+def _treat_path_dracut(dmodule):
     """
     In case the path is not set, set the expected path of the dracut module.
     """
+
     if not dmodule.module_path:
         return os.path.join(DRACUT_MOD_DIR, dmodule.name)
     return dmodule.module_path
 
 
-def _detect_dracut_modules_conflicts(msgtype):
+def _treat_path_kernel(kmodule):
+    """
+    In case the path of a kernel module is not set, indicate that the module is
+    taken from the current system.
+    """
+
+    if not kmodule.module_path:
+        return kmodule.name + ' (system)'
+    return kmodule.module_path
+
+
+def _detect_modules_conflicts(msgtype, kind):
     """
     Return dict of modules with conflicting tasks
 
-    In this case when a dracut module should be applied but different
-    sources are specified. E.g.:
-       include dracut modules X where,
+    In this case when a module should be applied but different sources are
+    specified. E.g.:
+       include modules X where,
          msg A)  X
          msg B)  X from custom path
     """
-    dracut_modules = defaultdict(set)
+
+    msgattr = None
```

Comment:
Nit: just a refactoring suggestion, keep any variant

I'd probably create a clearly defined mapping then set couple of same attributes via if/else. But totally up to you, as long as there are only 2 options any approach should be readable.
```
modules_map = {'dracut': {'msgattr': 'include_dracut_modules', 'treat_path_func': _treat_path_dracut,
               'kernel': {'msgattr': 'include_kernel_modules', 'treat_path_func': _treat_path_kernel }}}
```
```
for msg in api.consume(msgtype):
    for module in getattr(msg, modules_map[kind]['msgattr']):
        modules[module.name].add(modules_map[kind]['treat_path_func'](module))
```

Reply:
I like this much more as well! Changed!